mpc/mpc-core/src/commit.rs

```diff
@@ -0,0 +1,124 @@
+//! This module provides a hash commitment scheme for types which implement `serde::Serialize`
+
+use crate::hash::{Hash, SecureHash};
+use rand::{thread_rng, Rng};
+use serde::{Deserialize, Serialize};
+
+/// Error associated with commitments
+#[derive(Debug, thiserror::Error)]
+#[allow(missing_docs)]
+pub enum CommitmentError {
+    #[error("Invalid decommitment")]
+    InvalidDecommitment,
+}
+
+/// A randomly generated 32 byte nonce
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+pub struct Nonce([u8; 32]);
```

Comment:
nonces prevent replay attacks. this is not why we add randomness to the hash. `Salt` is a more fitting name here.